### PR TITLE
SSL root cert support

### PR DESCRIPTION
**Pull request by comphead**

### CODE REVIEW

**Comment by nooberfsh** [comment]
What about this:
```rust
pub struct Certificate(pub (crate) reqwest::Certificate);

impl Certificate {
    pub fn read_pem<P: AsRef<Path>>(p: P) -> Result<Self> {...}  
    pub fn read_der<P: AsRef<Path>>(p: P) -> Result<Self> {...}
}

impl ClientBuilder {
    pub fn add_root_certificate(self, cert: Certificate) -> ClientBuilder {...}
}
```

**Comment by comphead** [comment]
> What about this:
> 
> ```rust
> pub struct Certificate(pub (crate) reqwest::Certificate);
> 
> impl Certificate {
>     pub fn read_pem<P: AsRef<Path>>(p: P) -> Result<Self> {...}  
>     pub fn read_der<P: AsRef<Path>>(p: P) -> Result<Self> {...}
> }
> 
> impl ClientBuilder {
>     pub fn add_root_certificate(self, cert: Certificate) -> ClientBuilder {...}
> }
> ```

Im wondering do we really need a thin wrapper over `reqwest::Certificate`? 

Reg ClientBuilder,root_cert is not really part of Client. Rather its good to have kind of SslBuilder, which can contain ssl related params like root_certificate, client certificate, ssl trustore, ssl trust store pwd, etc. I can do ssl builder as part of another PR to keep this readable and relatively small

File: src/ssl.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+}
+
+impl Ssl {
+    pub fn cert_from_pem<P: AsRef<Path>>(root_certificate_path: P) -> Result<reqwest::Certificate> {
+        let mut buf = Vec::new();
+        std::fs::File::open(root_certificate_path.as_ref())
+            .unwrap()
+            .read_to_end(&mut buf)
+            .unwrap();
```
Comment (nooberfsh) [inline src/ssl.rs:25]: we should really avoid `unwrap`.

File: src/ssl.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        std::fs::File::open(root_certificate_path)
+            .expect(&format!(
+                "Error opening certificate file {}",
+                root_certificate_path
+            ))
+            .read_to_end(&mut buf)
+            .expect(&format!(
+                "Error reading certificate file {}",
+                root_certificate_path
+            ));
```
Comment (nooberfsh) [inline src/ssl.rs:43]: we should really avoid expect.

**Pull request closed without merging**